### PR TITLE
Constraint setuptools to 72.1.0 or older in requirements.txt

### LINKED ISSUE #136541: AttributeError: module 'distutils' has no attribute '_msvccompiler'
### 🐛 Describe the bug

When trying to build PyTorch from source with the latest setuptools>=74.0.0 due to some of the internal private functions from distutils being used as part of the build and are longer monkeypatched starting from 74.0.0 https://github.com/pytorch/pytorch/blob/274883083d99511d33a6b75af40c99f7613fe40d/tools/build_pytorch_libs.py#L37

```
python setup.py bdist_wheel

Traceback (most recent call last):
File "<stdin>", line 1, in <module>
AttributeError: module 'distutils' has no attribute '_msvccompiler'
```
Also observed in CI jobs: https://github.com/pytorch/pytorch/actions/runs/10979326524

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.29.0-rc3
Libc version: N/A

Python version: 3.12.6 | packaged by conda-forge | (main, Sep 22 2024, 14:01:26) [MSC v.1941 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.22631-SP0
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture=9
CurrentClockSpeed=3000
DeviceID=CPU0
Family=207
L2CacheSize=32768
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=3000
Name=13th Gen Intel(R) Core(TM) i9-13900K
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] numpy==2.1.1
[pip3] optree==0.12.1
[conda] numpy                     2.1.1                    pypi_0    pypi
[conda] optree                    0.12.1                   pypi_0    pypi

cc @malfet @seemethere @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex

### PR DESCRIPTION
**Pull request by ratnampa**
FIXES: https://github.com/pytorch/pytorch/issues/136541

Setuptools>=74.0.0 has deprecated support for some functions in distutils, and so the builds run into error such as ```AttributeError: module 'distutils' has no attribute '_msvccompiler'```. Also, the pytorch builds have setuptools pin to 72.1.0 according to these PRs: https://github.com/pytorch/builder/pull/1995 and https://github.com/pytorch/builder/commit/89d9a8cf6f4357d5d523b8d7c46351ac913e1309. So, until there is a fix to change the function usage in accordance with latest setuptools, the 72.1.0 version works fine. 

Also observed in CI jobs: https://github.com/pytorch/pytorch/actions/runs/10979326524

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -18,6 +18,6 @@ fsspec
 lintrunner
 ninja
 # setuptools was removed from default python install
-setuptools ; python_version >= "3.12"
+setuptools==72.1.0 ; python_version >= "3.12"
```
Comment (malfet) [inline requirements.txt:21]: Unsure why those 3 lines are needed, please consider removing

File: requirements.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ numpy
 psutil
 pyyaml
 requests
-setuptools
+setuptools==72.1.0
```
Comment (malfet) [inline requirements.txt:9]: Please add a comment why one needs to specify an explicit version here? And wouldn't `<=72.1.0` be better?

**Review by malfet — CHANGES_REQUESTED** [review]
Thank you for your PR. Please rebase your change(it's 1000+ commits behind latest trunk) to get a clear signal.And also add a comment to requirements.txt to explain why this is needed

File: requirements.txt (hunk 1 above)
Comment (ratnampa) [reply requirements.txt]: Done

File: requirements.txt (hunk 2 above)
Comment (ratnampa) [reply requirements.txt]: Done

**Review by malfet — APPROVED** [review]

File: requirements.txt [hunk 3]
```diff
@@ -6,7 +6,10 @@ numpy
 psutil
 pyyaml
 requests
-setuptools
+# Setuptools>=74.0.0 stopped support for directly using private funcs(_msvccompiler)
+# and consolidated all compiler logic in distutils used in Pytorch build, so older
```
[the remark this answers is not in the document]
Comment (ratnampa) [reply requirements.txt]: Thank you!

**Pull request closed without merging**